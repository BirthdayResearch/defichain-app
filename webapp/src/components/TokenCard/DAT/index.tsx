import React from 'react';
import { Card, CardBody, Col, Row } from 'reactstrap';
import { I18n } from 'react-redux-i18n';

import styles from '../TokenCard.module.scss';
import { ITokenCard } from '../../../utils/interfaces';

import TokenAvatar from '../../TokenAvatar';

const TokenCard: React.FunctionComponent<ITokenCard> = (props: ITokenCard) => {
  const { data } = props;

  return (
    <Card
      className={styles.tokenCard}
      onClick={() => props.handleCardClick(data.symbol, data.hash)}
    >
      <CardBody className={styles.cardBody}>
        <Row className='mb-2'>
          <Col md='8' className='d-flex align-items-center'>
            <div>
              <span className={styles.tokenName}>
                {data.isLPS
                  ? `${I18n.t(
                      'containers.tokens.tokensPage.datLabels.datLabels'
                    )} ${data.symbolKey}`
                  : data.name}
              </span>{' '}
              <span className={styles.tokenSymbol}>{data.symbolKey}</span>
            </div>
          </Col>
          {
            <Col md='4' className={`${styles.tokenCardIcon}`}>
              <TokenAvatar
                symbol={data.symbolKey}
                size='32px'
                textSizeRatio={2}
              />
            </Col>
          }
        </Row>
        <Row>
          <Col className={styles.label}>
            {I18n.t('containers.tokens.tokensPage.datLabels.type')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {data.isDAT ? 'DAT' : ''}
          </Col>
        </Row>
        <Row>
          <Col className={styles.label}>
            {/* {I18n.t('containers.tokens.tokensPage.datLabels.price')} */}
            {I18n.t('containers.tokens.tokensPage.datLabels.decimal')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {data.decimal}
            {/* {data.price} */}
          </Col>
        </Row>
        <Row>
          <Col className={styles.label}>
            {/* {I18n.t('containers.tokens.tokensPage.datLabels.volume')} */}
            {I18n.t('containers.tokens.tokensPage.datLabels.limit')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {/* {data.volume} */}
            {data.limit}
          </Col>
        </Row>
        <Row>
          <Col className={styles.label}>
            {/* {I18n.t('containers.tokens.tokensPage.datLabels.marketCap')} */}
            {I18n.t('containers.tokens.tokensPage.datLabels.mintingSupport')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {/* {data.marketCap} */}
            {data.mintable.toString()}
          </Col>
        </Row>
        <Row>
          <Col className={styles.label}>
            {/* {I18n.t('containers.tokens.tokensPage.datLabels.holders')} */}
            {I18n.t('containers.tokens.tokensPage.datLabels.tradeable')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {/* {data.holders} */}
            {data.tradeable.toString()}
          </Col>
        </Row>
      </CardBody>
    </Card>
  );
};

export default TokenCard;
