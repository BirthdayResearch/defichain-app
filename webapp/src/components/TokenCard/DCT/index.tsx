import React from 'react';
import { Card, CardBody, Col, Row } from 'reactstrap';
import { I18n } from 'react-redux-i18n';

import styles from '../TokenCard.module.scss';
import { ITokenCard } from '../../../utils/interfaces';
import VerifiedIcon from '../../../assets/svg/verified-icon.svg';

import TokenAvatar from '../../TokenAvatar';

const TokenCard: React.FunctionComponent<ITokenCard> = (props: ITokenCard) => {
  const { data } = props;

  return (
    <Card
      className={styles.tokenCard}
      onClick={() => props.handleCardClick(data.symbol, data.hash)}
    >
      <CardBody className={styles.cardBody}>
        <Row className='mb-2'>
          <Col md='8' className='d-flex align-items-center'>
            <div>
              <span className={styles.tokenName}>
                {data.isLPS
                  ? `${I18n.t(
                      'containers.tokens.tokensPage.dctLabels.liquidityTokenFor'
                    )} ${data.symbolKey}`
                  : data.name}
              </span>{' '}
              <span className={styles.tokenSymbol}>{data.symbolKey}</span>
            </div>
          </Col>
          <Col md='4' className={`${styles.tokenCardIcon}`}>
            {data.isDAT && (
              <img src={VerifiedIcon} className={styles.verifiedIcon} />
            )}
            <TokenAvatar
              symbol={data.symbolKey}
              size='32px'
              textSizeRatio={2}
            />
          </Col>
        </Row>
        {/* <Row>
          <Col className={styles.label}>
            {I18n.t('containers.tokens.tokensPage.dctLabels.type')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {data.isDAT ? '' : 'DCT'}
          </Col>
        </Row> */}
        <Row>
          <Col className={styles.label}>
            {/* {I18n.t(
              'containers.tokens.tokensPage.dctLabels.totalInitialSupply'
            )} */}
            {I18n.t('containers.tokens.tokensPage.dctLabels.limit')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {/* {data.totalInitialSupply} */}
            {data.limit}
          </Col>
        </Row>
        <Row>
          <Col className={styles.label}>
            {/* {I18n.t('containers.tokens.tokensPage.dctLabels.finalSupplyLimit')} */}
            {I18n.t('containers.tokens.tokensPage.dctLabels.decimal')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {/* {data.finalSupplyLimit} */}
            {data.decimal}
          </Col>
        </Row>
        <Row>
          <Col className={styles.label}>
            {I18n.t('containers.tokens.tokensPage.dctLabels.mintingSupport')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {data.mintable.toString()}
          </Col>
        </Row>
        <Row>
          <Col className={styles.label}>
            {I18n.t('containers.tokens.tokensPage.dctLabels.tradeable')}
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>
            {data.tradeable.toString()}
          </Col>
        </Row>
      </CardBody>
    </Card>
  );
};

export default TokenCard;
