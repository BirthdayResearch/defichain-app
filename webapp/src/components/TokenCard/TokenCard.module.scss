// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../scss/variables';
@import '../../scss/bootstrap/variables';

.tokenCard {
  margin-bottom: 24px;
  cursor: pointer;
  height: calc(100% - 24px);
}

.tokenBalanceCard {
  margin-bottom: 24px;
}

.text {
  text-align: end;
}

.cardBody {
  display: inline-block;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: flex-end;
}

.tokenName {
  font-weight: $font-weight-medium;
}

.tokenSymbol {
  color: $secondary;
}

.label {
  letter-spacing: 0.15px;
  width: 100%;
}

.value {
  font-size: 2.125rem;
  line-height: 2.5rem;
}

.unit {
  letter-spacing: 0.15px;
  color: $secondary;
}

.verifiedIcon {
  margin-right: 8px;
}

.tokenCardIcon {
  align-self: flex-start;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.keyValueLi {
  display: flex;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;
  padding: $table-cell-padding-y 0;
}

.cardValue {
  display: flex;
  color: $secondary;
}

.cardLabel {
  color: $secondary;
  padding-right: 1rem;
}

.cursor {
  cursor: pointer;
}
