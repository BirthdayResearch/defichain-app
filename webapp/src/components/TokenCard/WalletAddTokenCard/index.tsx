import React from 'react';
import { Card, CardBody, Col, Row, Button } from 'reactstrap';
import { I18n } from 'react-redux-i18n';

import styles from '../TokenCard.module.scss';
import { IWalletTokenCard } from '../../../utils/interfaces';
// import Icon from '../../../assets/svg/icon-coin-bitcoin-lapis.svg';

interface WalletAddTokenCardProps {
  token: IWalletTokenCard;
}

const WalletAddTokenCard: React.FunctionComponent<WalletAddTokenCardProps> = (
  props: WalletAddTokenCardProps
) => {
  const { token } = props;
  return (
    <Card className={styles.tokenCard}>
      <CardBody className={styles.cardBody}>
        <Row>
          <Col md='6'>
            <div className='d-flex'>
              {/* <img src={Icon} /> */}
              <div className='ml-4'>
                <div>
                  <b>{token.symbol}</b>
                </div>
                <div className={styles.cardValue}>{token.name}</div>
              </div>
            </div>
          </Col>
          <Col md='6'>
            <div className='float-right'>
              <Button color='link'>
                <span className='d-lg-inline'>
                  {I18n.t('containers.wallet.walletAddTokensPage.add')}
                </span>
              </Button>
            </div>
          </Col>
        </Row>
      </CardBody>
    </Card>
  );
};

export default WalletAddTokenCard;
