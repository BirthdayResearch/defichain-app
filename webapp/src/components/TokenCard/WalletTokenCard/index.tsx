import React from 'react';
import {
  Card,
  CardBody,
  Col,
  DropdownItem,
  DropdownMenu,
  DropdownToggle,
  Row,
  UncontrolledDropdown,
} from 'reactstrap';
import { I18n } from 'react-redux-i18n';

import styles from '../TokenCard.module.scss';
import { IWalletTokenCard } from '../../../utils/interfaces';

import TokenAvatar from '../../TokenAvatar';
import NumberMask from '../../NumberMask';
import BigNumber from 'bignumber.js';
import { MdMoreHoriz, MdRemove } from 'react-icons/md';
import {
  handleCheckToken,
  handleRemoveToken,
} from '../../../containers/WalletPage/service';

interface WalletTokenCardProps {
  token: IWalletTokenCard;
  handleCardClick: (
    symbol: string | null,
    hash: string,
    amount: any,
    address: string,
    isLPS: boolean
  ) => void;
  setwalletTableData?: any;
}

const WalletTokenCard: React.FunctionComponent<WalletTokenCardProps> = (
  props: WalletTokenCardProps
) => {
  const { handleCardClick, token, setwalletTableData } = props;

  return (
    <Row className='align-items-center'>
      <Col md='12'>
        <Card className={styles.tokenBalanceCard}>
          <CardBody className={styles.cardBody}>
            <Row className='align-items-center'>
              <Col
                md='4'
                className={styles.cursor}
                onClick={() =>
                  handleCardClick(
                    token.symbol,
                    token.hash,
                    token.amount,
                    token.address,
                    token.isLPS ?? false
                  )
                }
              >
                <div className='d-flex align-items-center justify-content-start'>
                  <div>
                    <TokenAvatar symbol={token.symbolKey} textSizeRatio={2} />
                  </div>
                  <div className='ml-4'>
                    <div>
                      <b>{token.symbolKey}</b>
                    </div>
                    <div className={styles.cardValue}>
                      {token.isLPS
                        ? `${I18n.t(
                            'containers.tokens.tokensPage.dctLabels.liquidityTokenFor'
                          )} ${token.symbolKey}`
                        : token.name}
                    </div>
                  </div>
                </div>
              </Col>
              <Col
                md='7'
                className={styles.cursor}
                onClick={() =>
                  handleCardClick(
                    token.symbol,
                    token.hash,
                    token.amount,
                    token.address,
                    token.isLPS ?? false
                  )
                }
              >
                <div className={`${styles.cardValue} justify-content-end`}>
                  <b className='text-dark'>
                    <NumberMask
                      value={new BigNumber(token.amount || 0).toFixed(8)}
                    />
                  </b>
                  <span className='ml-2'>{token.symbolKey}</span>
                </div>
              </Col>
              <Col md='1'>
                <UncontrolledDropdown
                  className={`${styles.cardValue} justify-content-end`}
                >
                  <DropdownToggle
                    className='padless'
                    color='link'
                    disabled={handleCheckToken(token)}
                  >
                    <MdMoreHoriz />
                  </DropdownToggle>
                  <DropdownMenu right>
                    <DropdownItem
                      className={styles.cursor}
                      onClick={() => {
                        const list = handleRemoveToken(token);
                        setwalletTableData(list);
                      }}
                    >
                      <MdRemove />
                      <span>
                        {I18n.t('containers.tokens.tokensPage.remove')}
                      </span>
                    </DropdownItem>
                  </DropdownMenu>
                </UncontrolledDropdown>
              </Col>
            </Row>
          </CardBody>
        </Card>
      </Col>
    </Row>
  );
};

export default WalletTokenCard;
