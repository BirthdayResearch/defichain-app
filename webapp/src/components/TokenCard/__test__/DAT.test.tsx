import React from 'react';
import TokenCard from '../DAT';
import { shallow } from 'enzyme';
import { toUpper } from 'lodash';

describe('TokenCard component', () => {
  it('should check for snapshot', () => {
    const data = {
      hash: 'hash',
      name: 'name',
      symbol: 'symbol',
      symbolKey: 'symbolKey',
      isDAT: true,
      decimal: 1,
      limit: 1,
      mintable: true,
      tradeable: true,
      isLPS: true,
    };
    const props = {
      data: data,
      handleCardClick: () => {},
    };
    const wrapper = shallow(<TokenCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
