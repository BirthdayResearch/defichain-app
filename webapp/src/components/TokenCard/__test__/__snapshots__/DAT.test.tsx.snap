// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenCard component should check for snapshot 1`] = `
<Card
  onClick={[Function]}
  tag="div"
>
  <CardBody
    tag="div"
  >
    <Row
      className="mb-2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        className="d-flex align-items-center"
        md="8"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div>
          <span>
            datLabels symbolKey
          </span>
           
          <span>
            symbolKey
          </span>
        </div>
      </Col>
      <Col
        className="undefined"
        md="4"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <TokenAvatar
          size="32px"
          symbol="symbolKey"
          textSizeRatio={2}
        />
      </Col>
    </Row>
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Type
      </Col>
      <Col
        className="undefined undefined"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        DAT
      </Col>
    </Row>
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Decimal
      </Col>
      <Col
        className="undefined undefined"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        1
      </Col>
    </Row>
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Limit
      </Col>
      <Col
        className="undefined undefined"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        1
      </Col>
    </Row>
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Minting support
      </Col>
      <Col
        className="undefined undefined"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        true
      </Col>
    </Row>
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Tradeable
      </Col>
      <Col
        className="undefined undefined"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        true
      </Col>
    </Row>
  </CardBody>
</Card>
`;
